### ISSUE TITLE
Allow propagation of clustering metadata

### ISSUE THREAD

**Issue by erooke** [zen-mapper/zen-mapper#33]
Some clustering algorithms produce other artifacts beyond the literal clustering. For example [affinity propagation](https://scikit-learn.org/1.5/modules/generated/sklearn.cluster.AffinityPropagation.html) produces cluster centers and an affinity matrix. Mapper itself does not care about this extra data but end users might.

At the moment the interface for a clusterer is specified by [`Clusterer`](https://github.com/zen-mapper/zen-mapper/blob/6dc0c859b542d8cfd050a0b032b5c4cfd8480582/src/zen_mapper/cluster.py#L10-L17) and it only allows for the cluster labels to be propagated. To support this use case this protocol is going to have to be redefined somehow.
